apps/s_server.c

```diff
@@ -3377,6 +3377,7 @@ static int www_body(int s, int stype, int prot, unsigned char *context)
  err:
     OPENSSL_free(buf);
     BIO_free_all(io);
+    BIO_free_all(ssl_bio);
```

Comment:
This isn't sufficient.  The `io` BIO chain includes this one, so they'll be double freed.  The correct fix is to set ssl_bio to NULL after the `BIO_push()` call.

Reply:
The NULL check isn't required.  All of our _free() calls accept a NULL argument safely.